src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/EnvironmentObjectList.java

```diff
@@ -220,6 +226,73 @@ public void update(int index,
               });
    }
    
+   private void createResizeColumn()
+   {
+      SafeHtmlRenderer<String> resizerRenderer =
+         new AbstractSafeHtmlRenderer<String>()
+         {
+            @Override
+            public SafeHtml render(String object)
+            {
+               SafeHtmlBuilder sb = new SafeHtmlBuilder();
+               sb.appendHtmlConstant(object);
+               return sb.toSafeHtml();
+            }
+         };
+      objectResizeColumn_ = new Column<RObjectEntry, String>(
+         new ClickableTextCell(resizerRenderer))
+         {
+            @Override
+            public String getValue(RObjectEntry object)
+            {
+               // We don't have ready access to the DOM (still under
+               // construction at this point), so we use a window method here.
+               return "<div class=\"" + style_.colResizer() + "\" " +
+                      "onmousedown=\"rstudio_beginResize(this, " +
+                      "\'" + style_.nameCol() + "'" +
+                      ");\"></div>";
+            }
+         };
+   }
+   
+   private final native void exportResizer() /*-{
+      $wnd.rstudio_beginResize = function(ele, clazz) {
+
+         // Locate the element to resize
+         var tbody = ele.parentElement.parentElement.parentElement;
+         var matches = tbody.firstElementChild.getElementsByClassName(clazz);
+         if (!matches)
+            return;
+         var resizer = matches[0];
+
+         // Initial position of cursor when resizing and initial width of the
+         // element
+         var start = -1;
+         var width = resizer.clientWidth;
+
+         // Resize method; given a mouse move event, adjust the size accordingly
+         var resize = function(evt) {
+            if (start === -1)
+               start = evt.clientX;
+            var newWidth = (width + (evt.clientX - start));
+            
+            // Enforce minimum/maximum width
+            if (newWidth < 40)
+               newWidth = 40;
+            else if (newWidth > 500)
+               newWidth = 500;
+               
+            resizer.style.width = newWidth + "px";
+         };
+
+         // Wire up event listeners
+         $wnd.addEventListener("mousemove", resize);
+         $wnd.addEventListener("mouseup", function(evt) {
```

Comment:
Does this event still get fired if the user drag the mouse outside of the window and then performs a mouse-up? (Do we need to be cautious and de-register the handle in other cases, e.g. if focus is lost or similar?)

Reply:
Yes, it does!